R/utils_render_common.R

```diff
@@ -327,18 +325,89 @@ perform_col_merge <- function(data,
       rows_to_format_idx <- setdiff(rows_to_format_idx, zero_rows_idx)
 
       body[rows_to_format_idx, mutated_column] <-
-        glue_gt(
-          list(
-            "1" = body[[mutated_column]][rows_to_format_idx],
-            "2" = body[[second_column]][rows_to_format_idx]
-          ),
-          pattern
-        ) %>%
-        as.character()
+        as.character(
+          glue_gt(
+            list(
+              "1" = body[[mutated_column]][rows_to_format_idx],
+              "2" = body[[second_column]][rows_to_format_idx]
+            ),
+            pattern
+          )
+        )
 
-    } else {
+    } else if (type == "merge_uncert" && length(col_merge[[i]]$vars) == 3) {
+
+      mutated_column <- col_merge[[i]]$vars[1]
+      lu_column <- col_merge[[i]]$vars[2]
+      uu_column <- col_merge[[i]]$vars[3]
+
+      pattern_equal <- col_merge[[i]]$pattern
+      sep <- col_merge[[i]]$sep
+
+      # Transform the separator text depending on specific
+      # inputs and the `context`
+      sep <-
+        sep %>%
+        context_dash_mark(context = context) %>%
+        context_plusminus_mark(context = context)
+
+      if (context == "html") {
+
+        pattern_unequal <-
+          paste0(
+            "<<1>><span style=\"margin-left: 0.1em;\">",
+            "<span class=\"gt_two_val_uncert\">",
+            "+<<3>><br>",
+            context_minus_mark(context = context), "<<2>>",
+            "</span></span>"
+          )
 
-      data_tbl <- dt_data_get(data = data)
+      } else if (context == "latex") {
+
+        pattern_unequal <- "$<<1>>^{+<<3>>}_{-<<2>>}$"
+
+      } else if (context == "rtf") {
+
+        pattern_unequal <- "<<1>>(+<<3>>, -<<2>>)"
+      }
+
+      # Determine rows where NA values exist
+      na_1_rows <- is.na(data_tbl[[mutated_column]])
+      na_lu_rows <- is.na(data_tbl[[lu_column]])
+      na_uu_rows <- is.na(data_tbl[[uu_column]])
+      na_lu_or_uu <- na_lu_rows | na_uu_rows
+      lu_equals_uu <- data_tbl[[lu_column]] == data_tbl[[uu_column]] & !na_lu_or_uu
```

Comment:
Decided to pass `NA`s straight though because it's the most noticeable and least invasive way to go. Plus, people can modify `NA` with `fmt_missing()` or `text_transform()` so the missing values may be meaningful/useful.